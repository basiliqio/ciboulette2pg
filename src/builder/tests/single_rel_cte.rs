use super::*;

#[test]
fn simple() {
    let mut builder = Ciboulette2PostgresBuilder::default();
    let rel_table = Ciboulette2PostgresTableSettings::new(
        Cow::Borrowed("id"),
        Cow::Borrowed("uuid"),
        Some(Cow::Borrowed("public")),
        Cow::Borrowed("favorite_color"),
    );
    let main_table = Ciboulette2PostgresTableSettings::new(
        Cow::Borrowed("id"),
        Cow::Borrowed("uuid"),
        Some(Cow::Borrowed("public")),
        Cow::Borrowed("cte_peoples"),
    );
    let store = gen_bag();
    let url = Url::parse("http://localhost/peoples").unwrap();
    let ciboulette_request = gen_req_create_people(&store, &url);
    builder
        .gen_select_cte_single_rel(
            &rel_table,
            store.get_type("favorite_color").unwrap(),
            ciboulette_request.query(),
            &main_table,
            "favorite_color",
        )
        .unwrap();

    let res = builder.build().unwrap();
    insta::assert_debug_snapshot!(res);
}

#[test]
fn sparse() {
    let mut builder = Ciboulette2PostgresBuilder::default();
    let rel_table = Ciboulette2PostgresTableSettings::new(
        Cow::Borrowed("id"),
        Cow::Borrowed("uuid"),
        Some(Cow::Borrowed("public")),
        Cow::Borrowed("favorite_color"),
    );
    let main_table = Ciboulette2PostgresTableSettings::new(
        Cow::Borrowed("id"),
        Cow::Borrowed("uuid"),
        Some(Cow::Borrowed("public")),
        Cow::Borrowed("cte_peoples"),
    );
    let store = gen_bag();
    let url = Url::parse("http://localhost/peoples?fields[favorite_color]=color").unwrap();
    let ciboulette_request = gen_req_create_people(&store, &url);
    builder
        .gen_select_cte_single_rel(
            &rel_table,
            store.get_type("favorite_color").unwrap(),
            ciboulette_request.query(),
            &main_table,
            "favorite_color",
        )
        .unwrap();

    let res = builder.build().unwrap();
    insta::assert_debug_snapshot!(res);
}

#[test]
fn empty_sparse() {
    let mut builder = Ciboulette2PostgresBuilder::default();
    let rel_table = Ciboulette2PostgresTableSettings::new(
        Cow::Borrowed("id"),
        Cow::Borrowed("uuid"),
        Some(Cow::Borrowed("public")),
        Cow::Borrowed("favorite_color"),
    );
    let main_table = Ciboulette2PostgresTableSettings::new(
        Cow::Borrowed("id"),
        Cow::Borrowed("uuid"),
        Some(Cow::Borrowed("public")),
        Cow::Borrowed("cte_peoples"),
    );
    let store = gen_bag();
    let url = Url::parse("http://localhost/peoples?fields[favorite_color]=").unwrap();
    let ciboulette_request = gen_req_create_people(&store, &url);
    builder
        .gen_select_cte_single_rel(
            &rel_table,
            store.get_type("favorite_color").unwrap(),
            ciboulette_request.query(),
            &main_table,
            "favorite_color",
        )
        .unwrap();

    let res = builder.build().unwrap();
    insta::assert_debug_snapshot!(res);
}
