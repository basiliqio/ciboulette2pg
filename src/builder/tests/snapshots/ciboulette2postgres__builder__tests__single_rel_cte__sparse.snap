---
source: src/builder/tests/single_rel_cte.rs
expression: res

---
(
    "SELECT \"public\".\"favorite_color\".\"id\"::TEXT AS \"id\", $0::TEXT AS \"type\", NULL::json  AS \"data\" FROM \"public\".\"favorite_color\" WHERE \"public\".\"favorite_color\".\"id\" IN (SELECT \"public\".\"cte_peoples\".\"favorite_color\" FROM \"public\".\"cte_peoples\")",
    Ciboulette2SqlArguments {
        inner: [
            Text(
                Some(
                    "favorite_color",
                ),
            ),
        ],
    },
)
